### PR TITLE
T27533: fix OS 2.6 -> 3.x upgrade script

### PR DESCRIPTION
**Pull request by ramcq**
Fix our upgrade script to fetch unexpired keys before pulling the new OS version. Also clean up a few other deprecated behaviours to better match what current 3.x expects.

https://phabricator.endlessm.com/T27533

### CODE REVIEW

**Comment by ramcq** [comment]
Untested! This script is fetched from git master in normal use, so do not merge until QA'd per https://phabricator.endlessm.com/T27533

File: eos-upgrade-eos2-to-eos3
Comment (wjt) [inline eos-upgrade-eos2-to-eos3]: I'm pretty sure from reading the code that the answer is yes, but I hope that this adds to the keyring rather than replacing its contents.

**Review by wjt — APPROVED** [review]
Approved, modulo confirmation that the test steps on https://phabricator.endlessm.com/T27533 actually work.

File: eos-upgrade-eos2-to-eos3
Comment (ramcq) [reply eos-upgrade-eos2-to-eos3]: Yes, I intentionally tested on my system with just EOSK1 and it was still able to verify a 3.6 commit that was only signed with EFSK1. So this per-repo keyring is additive to the platform trusted keys.

**Comment by wjt** [comment]
That change looks fine too.

**Pull request merged**
